src/Symfony/Component/DependencyInjection/Compiler/AttributeAutoconfigurationPass.php

```diff
@@ -50,8 +53,25 @@ public function process(ContainerBuilder $container): void
                     $configurator($conditionals, $attribute->newInstance(), $reflector);
                 }
             }
+
+            if ($constructor = $reflector->getConstructor()) {
```

Comment:
This should use the logic in [`AbstractRecursivePass::getConstructor()`](https://github.com/symfony/symfony/blob/5.x/src/Symfony/Component/DependencyInjection/Compiler/AbstractRecursivePass.php#L111)

Also, public properties and method calls should be analyzed I guess (see `AbstractRecursivePass::getReflectionMethod()`)

What about nested inline definitions btw?

Reply:
> This should use the logic in [`AbstractRecursivePass::getConstructor()`](https://github.com/symfony/symfony/blob/5.x/src/Symfony/Component/DependencyInjection/Compiler/AbstractRecursivePass.php#L111)

That would mean, we would extend the logic to factory methods, right? That probably makes sense.

> Also, public properties and method calls should be analyzed I guess (see `AbstractRecursivePass::getReflectionMethod()`)
>
> What about nested inline definitions btw?

I haven't taken nested definitions into account yet. But I agree, it would probably we odd if this feature would not work there. I'll look into this.